pyfar/classes/audio.py

```diff
@@ -166,7 +166,23 @@ def __getitem__(self, key):
 
 
         """
-        data = self._data[key]
+
+        # add empty slice at the end to always get all data contained in last
+        # dimension (samples or frequency bins)
+        if hasattr(key, '__iter__'):
+            key = (*key, slice(None))
+
+        # try indexing and raise verbose errors if it fails
+        try:
+            data = self._data[key]
```

Comment:
Unfortunately, this still does not work for all cases:
When accessing an object with more than one channel dimension, and accessing a single slice of a channel, this dimension is squeezed. I wrote a unit test (see `test_audio_frequency_data.py`), which invokes this error.